db/memtable_list.cc

```diff
@@ -455,20 +491,27 @@ Status MemTableList::TryInstallMemtableFlushResults(
 }
 
 // New memtables are inserted at the front of the list.
-void MemTableList::Add(MemTable* m, autovector<MemTable*>* to_delete) {
+void MemTableList::Add(MemTable* m, autovector<MemTable*>* to_delete,
+                       size_t usage) {
   assert(static_cast<int>(current_->memlist_.size()) >= num_flush_not_started_);
   InstallNewVersion();
   // this method is used to move mutable memtable into an immutable list.
   // since mutable memtable is already refcounted by the DBImpl,
   // and when moving to the imutable list we don't unref it,
   // we don't have to ref the memtable here. we just take over the
   // reference from the DBImpl.
-  current_->Add(m, to_delete);
+  current_->Add(m, to_delete, usage);
   m->MarkImmutable();
   num_flush_not_started_++;
   if (num_flush_not_started_ == 1) {
     imm_flush_needed.store(true, std::memory_order_release);
   }
+  UpdateMemoryUsageExcludingLast();
+}
+
+void MemTableList::TrimHistory(autovector<MemTable*>* to_delete, size_t usage) {
+  current_->TrimHistory(to_delete, usage);
```

Comment:
We are not creating a new version, but changing current_? This breaks the assumption that version should be immutable after installed.

Reply:
Right now the caller need to call `InstallNewVersion`, I can move it to `TrimHistory`

Reply:
I don't understand the comment.

Reply:
Pls ignore my previous comment, added call to `InstallNewVersion()` in the latest commit